src/components/Inputs/TextField.tsx

```diff
@@ -0,0 +1,25 @@
+import { observer } from "mobx-react-lite";
+import "../Inputs/TextField.scss"
+
+interface ITextFieldProps {
```

Comment:
Uporabi genericen tip tukaj string ali number. Tega typa je value in poskrbi da bo onChange argument tudi

Reply:
Tudi to sem pol komaj začel razmišljat, ko sem poslal že... Glede na to da lahko sprejme "text: string" ali pa "number: number"

Reply:
Tako. Se bolse mogoce, probaj nareti na type. Torej ce je type: "text" pol ves da je value string, type "number" ves da je number in tako naprej. Ce ima textfield type text in mu das value number, more bit error pri value number

Reply:
Tudi o tem sem razmisljal... še boljša rešitev.

Reply:
Bi lahko to naredo z ternary operatorom v funckiji "handleChange" al sem si potem preveč zakompliciral?

Reply:
Probaj naret pa boma pol predlagala boljse resitve

Reply:
Pazi pa da interface prav sestavis in pa on change logiko

Reply:
`
import { observer } from "mobx-react-lite";
import "../Inputs/TextField.scss"

interface ITextFieldProps {
    onChange?: (value: string | number) => void,
    placeholder?: string
    value?: string | number
    className?: string
    type?: "text" | "number" | "date" | "file"
}

export const TextField: React.FC<ITextFieldProps> = observer(({ onChange, placeholder, value, className, type }) => {
    

    const validateType = (inputValue: string | number) => {
        if ((type === "number" && typeof inputValue !== "number") || (type === "text" && typeof inputValue !== "string")) {
            throw new Error(`Invalid type for 'value' prop in TextField. Expected type ${type}, got ${typeof inputValue}`);
        }
    };
    
    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const inputValue = e.target.value
        validateType(inputValue)
        onChange && onChange(inputValue)
    }
    
    return (
        <input
            type={type}
            placeholder={placeholder}
            onChange={handleChange}
            value={value}
            className={`base-input ${className}`} />
    )
})

TextField.defaultProps = {
    type: "text",
}

`

Do zaj mam to. Problem je endino, da mi javi error, ko vnesem v input "type: "number"" stevilke mi vrze error. Pomojem za to ker on vseeno prebere value kot string, tak da se poskusam to resit zaj.

Reply:
Okej. Zdaj boma prvo type porihtala. To je malo bolj advanced zadeva, tak da nič hudega. Če neboš razumel se ne sekirat, probaj naštudirat zakaj zdaj to dela. Vglavnem z to kodo si pomagaj:

```
type TextFieldType = "text" | "number"

type TextFieldValueType<T extends TextFieldType> = T extends "number" ? number : string

interface ITextFieldProps<T extends TextFieldType = "text"> {
	onChange?: (value: TextFieldValueType<T>) => void
	placeholder?: string
	value?: TextFieldValueType<T>
	className?: string
	type?: T
}

const TextField = <T extends TextFieldType = "text">({ onChange, placeholder, value, className, type = "text" }: ITextFieldProps<T>) => {
	// ...tvoja komponenta
}
```

Zdaj probaj si kopirat, probaj dat različen type, value, etc. pa bi moral dobiti error message če type maš recimo type text in value number kot prope za komponento.

Nadgradi logiko da podpreš še druge type, in da onChange vrneš pravi type.